### ISSUE TITLE
Add API v3 endpoints /analyses GET and /analyses/{id}/ GET

### ISSUE THREAD

**Issue by adrian-lara** [SEED-platform/seed#2440]
# /analyses
### Permissions
Org member

### Body
None

### Request Type
GET

### Params
- property_id (filter Analysis objects by this) - optional
- organization_id

### Incorporate logic
- Include summarizing each set of AnalysisRun statuses
    - This needs to be built out (likely a function on the Analysis model). For now, this can be as simple as returning the “lowest” enum status. (e.g. out of Completed, Running, Stopped, "Running" would be returned as it has a enum value of 4 against Completed, 7, and Stopped, 6)
- Append all cycle_ids to results (query -Runs and return unique cycle_ids) (likely a function on the Analysis model)
- Append count of associated AnalysisRuns (likely a function on the Analysis model)

### Returns
JSON with status and data (Analysis[])

# /analyses/{id}
### Permissions
Org member

### Body
none

### Request method
GET

### Query Params
analysis_id
organization_id (query param)

### Incorporate logic (same as /analyses - GET)
- Include summarizing each set of AnalysisRun statuses
    - This needs to be built out (likely a function on the Analysis model). For now, this can be as simple as returning the “lowest” enum status. (e.g. out of Completed, Running, Stopped, "Running" would be returned as it has a enum value of 4 against Completed, 7, and Stopped, 6)
- Append all cycle_ids to results (query -Runs and return unique cycle_ids) (likely a function on the Analysis model)
- Append count of associated AnalysisRuns (likely a function on the Analysis model)

### Returns
JSON with status and data (Analysis object)

**Comment by Ryoken** [comment]
The Analysis API is supposed to return all the relevant cycle IDs, which are stored on the AnalysisPropertyViews.  The ticket would like to see this functionality live on the Analysis model, but AnalysisPropertyViews already inherits Analysis in the model, so a circular dependency is introduced.

**Comment by Ryoken** [comment]
Previously, there was an AnalysisRun that had a status on it, so the Analysis endpoint was to collect and report the lowest status.  However, status is now on the Analysis model.  Is there no longer a need to aggregate status?

**Comment by Ryoken** [comment]
Does the /analyses/{id} still need an organization id?  An organization is required on each Analysis.

**Comment by adrian-lara** [comment]
> The Analysis API is supposed to return all the relevant cycle IDs, which are stored on the AnalysisPropertyViews. The ticket would like to see this functionality live on the Analysis model, but AnalysisPropertyViews already inherits Analysis in the model, so a circular dependency is introduced.

I don't think this would require adding any more relationships, so I'd imagine we'd avoid any circular dependencies. From what I'm imagining, we could query for all the Cycles of an Analysis via a left join between Analysis and  AnalysisPropertyViews. I might be misunderstanding something though.


> Previously, there was an AnalysisRun that had a status on it, so the Analysis endpoint was to collect and report the lowest status. However, status is now on the Analysis model. Is there no longer a need to aggregate status?

Yeah - for now, we don't need an aggregate status, so we can remove that endpoint.


> Does the /analyses/{id} still need an organization id? An organization is required on each Analysis so the information is redundant and potentially conflicting.

TL;DR: it's probably not entirely necessary, but let's just keep it for now.

Good question - I could be convinced either way here. @macintoshpie feel free to chime in. We tend to throw the organization_id query param on a lot of endpoints. I think the main reason is that users can have multiple organizations, so at the very least, in situations like this where we're querying for a single resource, this serves a "double check" that we're pulling a resource for the currently set organization (set in the top right of the UI).

**Comment by macintoshpie** [comment]
> Does the /analyses/{id} still need an organization id? An organization is required on each Analysis so the information is redundant and potentially conflicting.

I think removing the org id query param is a good idea for POST requests. As long as the body key is `organization_id` it should be usable by our auth middleware, and removes the redundancy/potential for confusion/conflict. If we didn't remove the query param, we'd need to verify it matches the body param b/c the query param is the first one used for auth. so removing it can make it clear which org id is being used.

For GET requests, we will need the query param b/c it won't be provided elsewhere.

**Comment by Ryoken** [comment]
https://github.com/SEED-platform/seed/pull/2473

**Comment by adrian-lara** [comment]
No manual testing needed.
